src/core.ts

```diff
@@ -478,6 +486,40 @@ function createPatchOuter<T>(
   }, patchConfig);
 }
 
+/**
+ * Creates a patcher that patches the document starting at node with a
+ * provided function. This function may be called during an existing patch operation.
+ * @param patchConfig The config to use for the patch.
+ * @returns The created function for patching an Element's children.
+ */
+function createPatchBetween<T, R>(patchConfig?: PatchConfig) {
+  return (startNode: Node, endNode: Node, template: (a: T | undefined) => void,
+  data?: T | undefined) => {
+    const patcher = createPatcher<T, R>((node, fn, data) => {
+      currentNode = startNode;
+      currentParent = startNode.parentNode;
+      currentEndNode = endNode;
+
+      fn(data);
+      if (DEBUG) {
+        if (currentNode === null) {
+          assertNoUnclosedTags(currentParent, endNode);
+        }
+      }
+      if (currentNode && currentNode.nextSibling !== endNode) {
```

Comment:
It should be possible to have unvisited dom but currentNode.nextSibling === endNode.

{template a}
<div>
</div>
{if $a}
  <div></div>
{else}
  <span></span>
{/if}
<div></div>

Reply:
Can you expand on that a bit? What would the unvisited DOM be? What are the idom opcode that produce it?

Reply:
It should be

function fn(a:boolean) {
  elementOpen('div', 1);
  elementClose();
  if (a) {
    elementOpen('div', 2);
    elementClose();
  } else {
    elementOpen('div', 3);
    elementClose();
  }
  elementOpen('div', 4);
  elementClose();
}

In the first route (A), you'd have

<div [key=1]></div>
 <div [key=2]></div>
<div [key=4]></div>

if you flip a to false, you'd first have, I think, the following behavior

<div [key=1]></div>
<div [key=3]></div> <-- insert
<div [key=2]></div>
<div [key=4]></div>

then the jump to 4
<div [key=1]></div>
<div [key=3]></div> <-- insert
<div [key=2]></div>
<div [key=4]></div> <--

and then never deleting [key=2]